torch/_inductor/scheduler.py

```diff
@@ -1916,6 +1943,145 @@ def create_scheduler_node(self, node: ir.Operation) -> BaseSchedulerNode:
         else:
             raise NotImplementedError(node)
 
+    def split_reduction_node(self) -> bool:
+        has_new_nodes = False
+
+        def split_reduction_single_node(idx, snode, all_nodes) -> None:
+            split_hints = []
+            for other_node in all_nodes:
```

Comment:
We shouldn't need to iterate over all_nodes. see, [here](https://github.com/pytorch/pytorch/blob/fd553b9817cb145e3cfaa52a08a831b9edf2fec7/torch/_inductor/scheduler.py#L2736-L2743) where we first group by buffer accesses.